Source/WebKit2/UIProcess/API/wpe/WPEWebAutomation.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2016 TATA ELXSI
+ * Copyright (C) 2016 Metrological
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef WPEWebAutomation_h
+#define WPEWebAutomation_h
+
+#include "APIObject.h"
+#include "WebInspectorProxy.h"
+#include "WPEWebAutomationClient.h"
+
+#include <JavaScriptCore/InspectorFrontendChannel.h>
+#include <JavaScriptCore/InspectorBackendDispatcher.h>
+#include <JavaScriptCore/InspectorFrontendRouter.h>
+#include <wtf/RefPtr.h>
+
+
+namespace WKWPE {
+class WebAutomation final: public API::ObjectImpl<API::Object::Type::AutomationSession>, public Inspector::FrontendChannel {
+
+public:
+    static WebAutomation* create();
+
+    void sendMessageToTarget(const String& command, void (*commandCallback)(WKStringRef rspMsg));
+    
+    ConnectionType connectionType() const override { return ConnectionType::Remote; }
+    bool sendMessageToFrontend(const String& rspMsg);
+    void setClient();
+    void setProcessPool(WebKit::WebProcessPool* processPool);
+    void setSessionIdentifier(const String& sessionIdentifier);
+
+private:
+    WebAutomation();
+    virtual ~WebAutomation();
+
+    bool m_connected;
```

Comment:
This seems unnecessary, you should be able to just check whether m_webAutomationSession is non-null.